File: Advanced_SQL_50/1755-defuse-the-bomb/defuse-the-bomb.ts
// TypeScript

function decrypt(code: number[], k: number): number[] {
    const N: number = code.length;
    const res: number[] = new Array(N).fill(0);
    
    for (let i = 0; i < N; i++) {
        if (k > 0) {
            for (let j = i + 1; j < i + 1 + k; j++) {
                res[i] += code[j % N];
            }
        } else if (k < 0) {
            for (let j = i - 1; j > i - 1 - Math.abs(k); j--) {
                res[i] += code[((j % N) + N) % N];
            }
        }
    }
    
    return res;
}

File: Advanced_SQL_50/1755-defuse-the-bomb/defuse-the-bomb.cs
public class Solution {
    public int[] Decrypt(int[] code, int k) {
        int N = code.Length;
        int[] res = new int[N];
        
        for (int i = 0; i < N; i++) {
            if (k > 0) {
                for (int j = i + 1; j < i + 1 + k; j++) {
                    res[i] += code[j % N];
                }
            } else if (k < 0) {
                for (int j = i - 1; j > i - 1 - Math.Abs(k); j--) {
                    res[i] += code[((j % N) + N) % N];
                }
            }
        }
        
        return res;
    }
}